pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -234,9 +282,11 @@ public void setBytesArray(int row, byte[][] bytesArray) {
 
   @Override
   public int getCharArray(int row, char[] charArray) {
-    int bufferIndex = headerReader.getInt(row, 1);
-    int startIndex = headerReader.getInt(row, 1);
-    int length = headerReader.getInt(row, 2);
+    FixedByteSingleValueMultiColReader headerReader = getCurrentReader(row);
+    int rowInCurrentHeader  = getRowInCurrentHeader(row);
+    int bufferIndex = headerReader.getInt(rowInCurrentHeader, 1);
```

Comment:
fixed for all the getters. 

This class is not used for anything other than int. Removing the other methods makes the logic crisp and clear, but because of the way inheritance is organized, I could not do it without significant changes (otherwise, that was my first choice).

I also considered adding a 'throws unsupported' on each of the others, but refrained at this time.

I will add a unit test for the other use cases for now, but probably not as extensive as the int[] use case.